examples/notebooks/mees_example.ipynb

```diff
@@ -0,0 +1,475 @@
+    "    layer_sizes=policy_layer_sizes,\n",
+    "    kernel_init=jax.nn.initializers.lecun_uniform(),\n",
+    "    final_activation=jnp.tanh,\n",
+    ")\n",
+    "\n",
+    "# Init population of controllers\n",
+    "random_key, subkey = jax.random.split(random_key)\n",
+    "fake_batch = jnp.zeros(shape=(1, env.observation_size))\n",
+    "keys = jnp.repeat(jnp.expand_dims(subkey, axis=0), repeats=1, axis=0)\n",
+    "init_variables = jax.vmap(policy_network.init)(keys, fake_batch)\n",
+    "\n",
+    "# Create the initial environment states for evaluating the samples\n",
+    "random_key, subkey = jax.random.split(random_key)\n",
+    "keys = jnp.repeat(jnp.expand_dims(subkey, axis=0), repeats=sample_number, axis=0)\n",
+    "reset_fn = jax.jit(jax.vmap(env.reset))\n",
+    "init_states_samples = reset_fn(keys)\n",
+    "\n",
+    "# Create the initial environment states for evaluating the final individual\n",
+    "random_key, subkey = jax.random.split(random_key)\n",
+    "keys = jnp.repeat(jnp.expand_dims(subkey, axis=0), repeats=1, axis=0)\n",
+    "init_states = reset_fn(keys)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "CZpBF24doQvF"
+   },
+   "source": [
+    "## Define the way the policy interacts with the env"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "XYeUDwTXoQvF"
+   },
+   "outputs": [],
+   "source": [
+    "# Define the fonction to play a step with the policy in the environment\n",
+    "def play_step_fn(\n",
+    "  env_state,\n",
+    "  policy_params,\n",
+    "  random_key,\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    Play an environment step and return the updated state and the transition.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    actions = policy_network.apply(policy_params, env_state.obs)\n",
+    "    \n",
+    "    state_desc = env_state.info[\"state_descriptor\"]\n",
+    "    next_state = env.step(env_state, actions)\n",
+    "\n",
+    "    transition = QDTransition(\n",
+    "        obs=env_state.obs,\n",
+    "        next_obs=next_state.obs,\n",
+    "        rewards=next_state.reward,\n",
+    "        dones=next_state.done,\n",
+    "        actions=actions,\n",
+    "        truncations=next_state.info[\"truncation\"],\n",
+    "        state_desc=state_desc,\n",
+    "        next_state_desc=next_state.info[\"state_descriptor\"],\n",
+    "    )\n",
+    "\n",
+    "    return next_state, policy_params, random_key, transition"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "AzNSJUUxoQvF"
+   },
+   "source": [
+    "## Define the scoring function and the way metrics are computed\n",
+    "\n",
+    "The scoring function is used in the evaluation step to determine the fitness and behavior descriptor of each individual. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "KeT2NuI_oQvG"
+   },
+   "outputs": [],
+   "source": [
```

Comment:
Is there no way to use the same scoring function to define the two scoring functions? To avoid the repetition that we have here.

Reply:
Yes it is feasible using the reset_based_scoring_function_brax_envs, I added it.